src/Tizen.Peripheral/Tizen.Peripheral/Spi.cs

```diff
@@ -0,0 +1,231 @@
+﻿/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeSpi = Interop.Peripheral.Spi;
+
+namespace Tizen.Peripheral
+{
+    /// <summary>
+    /// The class allows applications to communicate via SPI platform's bus.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    public class Spi : IDisposable
+    {
+        /// <summary>
+        /// Enumeration of SPI transfer modes.
+        /// </summary>
+        public enum Mode
+        {
+            /// <summary>
+            /// CPOL = 0, CPHa = 0 Mode.
+            /// </summary>
+            Mode0 = 0,
+
+            /// <summary>
+            /// CPOL = 0, CPHa = 1 Mode.
+            /// </summary>
+            Mode1,
+
+            /// <summary>
+            /// CPOL = 1, CPHa = 0 Mode.
+            /// </summary>
+            Mode2,
+
+            /// <summary>
+            /// CPOL = 1, CPHa = 1 Mode.
+            /// </summary>
+            Mode3
+        }
+
+        /// <summary>
+        /// Enumeration of bit orders.
+        /// </summary>
+        /// <remarks>
+        /// Currently only LSB order is supported!
+        /// </remarks>
+        public enum Order
+        {
+            /// <summary>
+            /// Use most siginificant bit first.
+            /// </summary>
+            MSB = 0,
+
+            /// <summary>
+            /// Use least siginificant bit first.
+            /// </summary>
+            LSB
+        }
+
+        //TODO Provide default values.
+        private Mode _transferMode;
+        private Order _bitOrder;
+        private byte _bitsPerWord;
+        private uint _frequency;
+
+        /// <summary>
+        /// Native handle to Spi.
+        /// </summary>
+        protected IntPtr _handle;
+
+        private Spi(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Opens a SPI slave device.
+        /// </summary>
+        /// <param name="bus">The SPI bus number.</param>
+        /// <param name="chip">The SPI chip select number.</param>
+        /// <returns></returns>
+        public static Spi Open(int bus, int chip)
```

Comment:
This `Open` Method is seen as a factory method.
However, what this method does is not different from what the constructor does.
Even if you don't have to use it as a factory like this, it's not different from just making it in the constructor using the same arguments as this method.

It seems that the factory is not needed, so it is recommended to just use the constructor.